fl4health/utils/load_data.py

```diff
@@ -1,85 +1,138 @@
+import random
 from logging import INFO
 from pathlib import Path
-from typing import Callable, Dict, Optional, Tuple, Union
+from typing import Callable, Dict, Optional, Tuple
 
+import torch
 import torchvision.transforms as transforms
 from flwr.common.logger import log
 from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10, MNIST
 
-from fl4health.utils.dataset import BaseDataset, Cifar10Dataset, MnistDataset
+from fl4health.utils.dataset import TensorDataset
 from fl4health.utils.dataset_converter import DatasetConverter
+from fl4health.utils.random import set_all_random_seeds
 from fl4health.utils.sampler import LabelBasedSampler
 
 
+def split_data_and_targets(
+    data: torch.Tensor, targets: torch.Tensor, validation_proportion: float = 0.02
+) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
+
+    total_size = data.shape[0]
+    train_size = int(total_size * (1 - validation_proportion))
+    train_indices = random.sample(range(total_size), train_size)
+    val_indices = [i for i in range(total_size) if i not in train_indices]
+    train_data, train_targets = data[train_indices, ...], targets[train_indices, ...]
```

Comment:
I'm not sure I've seen the `,...` notation before? I assume you're just saying take the whole of the tensor indexed by `train_indices`. Is there a reason to use that over just doing `data[train_indices]` and `targets[train_indices]`?